src/Controller/Component/PrgComponent.php

```diff
@@ -2,26 +2,75 @@
 namespace Search\Controller\Component;
 
 use Cake\Controller\Component;
-use Cake\Event\Event;
 
 class PrgComponent extends Component
 {
+
+    /**
+     * Default config
+     *
+     * ### Options
+     * - `actions` Method name(s) of actions to use PRG. Or bool for all or none.
+     *   You can pass a single action as string or multiple as array. Default is
+     *   true and all actions will be processed by the component.
+     *
+     * @var array
+     */
+    protected $_defaultConfig = [
+        'actions' => true,
+    ];
+
     /**
      * Checks if the current request has posted data and redirects the users
      * to the same action after converting the post data into GET params
      *
-     * @return void|Cake\Network\Response
+     * @return void|\Cake\Network\Response
      */
     public function startup()
     {
-        $controller = $this->_registry->getController();
-        $request = $controller->request;
+        $this->controller = $this->_registry->getController();
```

Comment:
The controller instance is redirected for using `Controller::redirect()` in `conversion()` so better to just use a local var in that method. I would prefer not adding extra properties.